backend/tests/clubs/test_views.py

```diff
@@ -2923,3 +2925,1035 @@ def test_club_approval_response_templates(self):
             content_type="application/json",
         )
         self.assertEqual(resp.status_code, 403)
+
```

Comment:
Appreciate the effort here but not sure if we need 1k lines of tests for this feature 😅 . From a quick glance, it looks like you're testing every action with every permission level – it's ok to omit testing everything except the most basic functionality.

Reply:
+1 on this. (In my opinion) unit tests are best when they're as contained but comprehensive as possible. (Here's a good read on this topic.) Just from a quick glance, I can see a lot of duplicate checks of the same functionality. There are 12 checks of the `membership-requests-list` route alone.

Reply:
As an example, I asked GPT to create a condensed version of the ownership requests-related tests. This might be a good starting point.
```python
def test_request_creation_permissions(self):
    """Test who can create ownership requests"""
    for user in [self.officer, self.member, self.non_member]:
        self.client.force_login(user)
        resp = self.client.post(
            reverse("ownership-requests-list"),
            {"club": self.club.code},
            content_type="application/json"
        )
        self.assertEqual(resp.status_code, 201)
        self.assertTrue(
            OwnershipRequest.objects.filter(
                club=self.club,
                requester=user
            ).exists()
        )
        # Verify email was sent
        self.assertEqual(len(mail.outbox), 1)
        mail.outbox.clear()

def test_request_viewing_permissions(self):
    """Test who can view ownership requests"""
    # Create a request
    request = OwnershipRequest.objects.create(
        club=self.club,
        requester=self.officer
    )

    # Test requester can view own request
    self.client.force_login(self.officer)
    resp = self.client.get(reverse("ownership-requests-list"))
    self.assertEqual(resp.status_code, 200)
    self.assertEqual(len(resp.json()), 1)

    # Test owner can view club requests
    self.client.force_login(self.owner)
    resp = self.client.get(
        reverse("club-ownership-requests-list", args=(self.club.code,))
    )
    self.assertEqual(resp.status_code, 200)
    self.assertEqual(len(resp.json()), 1)

    # Test non-owners cannot view club requests
    for user in [self.officer, self.member, self.non_member]:
        self.client.force_login(user)
        resp = self.client.get(
            reverse("club-ownership-requests-list", args=(self.club.code,))
        )
        self.assertEqual(resp.status_code, 403)

def test_request_withdrawal(self):
    """Test request withdrawal functionality"""
    request = OwnershipRequest.objects.create(
        club=self.club,
        requester=self.officer
    )

    self.client.force_login(self.officer)
    resp = self.client.delete(
        reverse("ownership-requests-detail", args=(self.club.code,))
    )
    self.assertIn(resp.status_code, [200, 204])

    # Verify request is marked withdrawn
    request.refresh_from_db()
    self.assertTrue(request.withdrawn)

    # Verify withdrawn request is not visible
    resp = self.client.get(
        reverse("ownership-requests-detail", args=(self.club.code,))
    )
    self.assertEqual(resp.status_code, 404)

def test_request_acceptance(self):
    """Test request acceptance functionality"""
    request = OwnershipRequest.objects.create(
        club=self.club,
        requester=self.officer
    )

    # Test non-owners cannot accept
    self.client.force_login(self.officer)
    resp = self.client.post(
        reverse("club-ownership-requests-accept", 
            kwargs={
                "club_code": self.club.code,
                "requester__username": self.officer.username
            }
        )
    )
    self.assertEqual(resp.status_code, 403)

    # Test owner can accept
    self.client.force_login(self.owner)
    resp = self.client.post(
        reverse("club-ownership-requests-accept",
            kwargs={
                "club_code": self.club.code,
                "requester__username": self.officer.username
            }
        )
    )
    self.assertEqual(resp.status_code, 200)

    # Verify membership was updated
    self.assertTrue(
        Membership.objects.filter(
            club=self.club,
            person=self.officer,
            role=Membership.ROLE_OWNER
        ).exists()
    )
```